bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/AdaptiveImageServlet.java

```diff
@@ -347,6 +347,12 @@ private void transformAndStreamAsset(SlingHttpServletResponse response, ValueMap
                     int resizeHeight = calculateResizeHeight(originalWidth, originalHeight, resizeWidth);
                     if (resizeHeight > 0 && resizeHeight != originalHeight) {
                         layer = getLayer(rendition);
+                        if (layer.getBackground().getTransparency() != Transparency.OPAQUE &&
+                                ("jpg".equalsIgnoreCase(extension) || "jpeg".equalsIgnoreCase(extension))) {
+                            LOGGER.debug("Adding default (white) background to a transparent PNG: {}/{}", asset.getPath(),
+                                    rendition.getName());
+                            layer.setBackground(Color.white);
```

Comment:
I wonder if the background color should be "Dark Mode aware" and prefer black bg over white when dark mode is enabled.

Reply:
We considered that but it's not feasible as the images will get cached at dispatcher level. We could introduce a selector but that's already too much overhead and we'd rather invest more in something like #496. Then you could have a dark/light rendition pair.

For now, if you need to switch between dark and light themes, best bet is to use an image format that supports transparency.